scripts/maintenance/standard-header-guards.pl

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/perl
+
+## Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+# source-maintenance plugin to standardize the header guards
+
+use strict;
+use warnings;
+
+my $filename = $ARGV[0];
+
+if ($filename !~ /\.h$/i || $filename =~ /^scripts\/boilerplate.h/i) {
+    while (<>) { print; }
+    exit(0);
+}
+
+my @accumulate=();
+my $current_guard = undef;
+
+my $first_ifdef_pos = undef;
+my $last_ifdef_pos = undef;
+while (<>) {
+    if (!defined($first_ifdef_pos) && /^#ifndef\s+(\w+_H)/ ) {
+        $current_guard = $1;
+        $first_ifdef_pos = $#accumulate+1;
+    } elsif (/^#endif/) {
+        $last_ifdef_pos = $#accumulate+1;
+    }
+    push(@accumulate, $_);
```

Comment:
Please push first, before the `if` statements above. Then you will not have to remember to compensate with `+1` inside those `if` statements...

Reply:
fair point. doing that